src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolDataViewModelSorter.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Windows.Data;
+using System.Windows.Markup;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    /// <summary>
+    /// Sorts immutable collections of <see cref="DocumentSymbolDataViewModel"/>s 
+    /// </summary>
+    internal sealed class DocumentSymbolDataViewModelSorter : MarkupExtension, IMultiValueConverter
+    {
+        public static DocumentSymbolDataViewModelSorter Instance { get; } = new();
+
+        public object Convert(object[] values, Type targetType, object? parameter, CultureInfo culture)
+        {
+            if (values[0] is ImmutableArray<DocumentSymbolDataViewModel> children &&
```

Comment:
If we assume there are always at least 2 items in the values, can we assume/throw if they aren't the types we expect?

Reply:
@jmarolf ?

Reply:
WPF will call us will `null` for the first value on initialization but there will always be two elements in the array. We could add an assert that throws if thats ever not the case though I we would need to file a P0 bug on the WPF team if that ever happened.